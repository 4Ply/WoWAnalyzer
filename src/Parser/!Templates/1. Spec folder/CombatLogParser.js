import CoreCombatLogParser from 'Parser/Core/CombatLogParser';

// TODO: Pick the ones relevant to your spec:
// import DamageDone from 'Parser/Core/Modules/DamageDone';
// import HealingDone from 'Parser/Core/Modules/HealingDone';
// import DamageTaken from 'Parser/Core/Modules/DamageTaken';

import Abilities from './Modules/Abilities';
import Checklist from './Modules/Features/Checklist';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';

// TODO: import CooldownThroughputTracker from './Modules/Features/CooldownThroughputTracker';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // TODO: Enable the ones relevant to your spec:
    // damageDone: [DamageDone, { showStatistic: true }],
    // healingDone: [HealingDone, { showStatistic: true }],
    // damageTaken: [DamageTaken, { showStatistic: true }],

    abilities: Abilities, // TODO: Go to Abilities class and fill it with all abilities available to your spec

    // Features
    checklist: Checklist, // TODO: Go into Checklist to define rules
    alwaysBeCasting: AlwaysBeCasting, // TODO: Define a list of all abilities on the GCD
    // TODO: cooldownThroughputTracker: CooldownThroughputTracker,
  };
}

export default CombatLogParser;
