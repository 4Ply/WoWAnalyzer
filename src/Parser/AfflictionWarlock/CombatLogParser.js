import React from 'react';

import SuggestionsTab from 'Main/SuggestionsTab';
import Tab from 'Main/Tab';
import Talents from 'Main/Talents';

import MainCombatLogParser from 'Parser/Core/CombatLogParser';

import CastEfficiency from './Modules/Features/CastEfficiency';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import CooldownTracker from './Modules/Features/CooldownTracker';
import AgonyUptime from './Modules/Features/AgonyUptime';
import CorruptionUptime from "./Modules/Features/CorruptionUptime";
import SiphonLifeUptime from "./Modules/Features/SiphonLifeUptime";

import SoulShardTracker from "./Modules/SoulShards/SoulShardTracker";
import SoulShardDetails from "./Modules/SoulShards/SoulShardDetails";

class CombatLogParser extends MainCombatLogParser {

  static specModules = {
    // Features
    castEfficiency: CastEfficiency,
    alwaysBeCasting: AlwaysBeCasting,
    cooldownTracker: CooldownTracker,

    // DoT uptimes
    agonyUptime: AgonyUptime,
    corruptionUptime: CorruptionUptime,
    siphonLifeUptime: SiphonLifeUptime,

    //Core
    soulShardTracker: SoulShardTracker,
    soulShardDetails: SoulShardDetails,
  };

  generateResults() {
    const results = super.generateResults();
    results.tabs = [
      {
        title: 'Suggestions',
        url: 'suggestions',
        render: () => (
          <SuggestionsTab issues={results.issues} />
        ),
      },
      {
        title: 'Talents',
        url: 'talents',
        render: () => (
          <Tab title="Talents">
            <Talents combatant={this.selectedCombatant} />
          </Tab>
        ),
      },
      // {
      //   title: 'Mana',
      //   url: 'mana',
      //   render: () => (
      //     <Tab title="Mana" style={{ padding: '15px 22px' }}>
      //       <Mana
      //         reportCode={this.report.code}
      //         actorId={this.playerId}
      //         start={this.fight.start_time}
      //         end={this.fight.end_time}
      //       />
      //     </Tab>
      //   ),
      // },
      ...results.tabs,
    ];

    return results;
  }
}

export default CombatLogParser;
