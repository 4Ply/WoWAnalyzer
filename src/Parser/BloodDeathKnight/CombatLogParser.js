import React from 'react';
import SuggestionsTab from 'Main/SuggestionsTab';
import Tab from 'Main/Tab';
import Talents from 'Main/Talents';
import CoreCombatLogParser from 'Parser/Core/CombatLogParser';

import CastEfficiency from './Modules/Features/CastEfficiency';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import CooldownTracker from './Modules/Features/CooldownTracker';
import BloodPlagueUptime from './Modules/Features/BloodPlagueUptime';
import BoneShieldUptime from './Modules/Features/BoneShieldUptime';
import OssuaryUptime from './Modules/Features/OssuaryUptime';
import WastedDeathAndDecay from './Modules/Features/WastedDeathAndDecay';
import RPWasted from './Modules/Features/RunicPowerWasted';
import BlooddrinkerTicks from './Modules/Features/BlooddrinkerTicks';

class CombatLogParser extends CoreCombatLogParser {

  static specModules = {

    //DeathKnight Core

    // Features
    castEfficiency: CastEfficiency,
    alwaysBeCasting: AlwaysBeCasting,
    cooldownTracker: CooldownTracker,
    BoneShieldUptime: BoneShieldUptime,
    OssuaryUptime: OssuaryUptime,
    WastedDeathAndDecay: WastedDeathAndDecay,
    RunicPowerWasted: RPWasted,
    BlooddrinkerTicks: BlooddrinkerTicks,

    //DOT
    bloodplagueUptime: BloodPlagueUptime,
    // Talents


    // Traits


    // Items:

  };

  generateResults() {
    const results = super.generateResults();

    results.tabs = [
      {
        title: 'Suggestions',
        url: 'suggestions',
        render: () => (
          <SuggestionsTab issues={results.issues} />
        ),
      },
      {
        title: 'Talents',
        url: 'talents',
        render: () => (
          <Tab title="Talents">
            <Talents combatant={this.modules.combatants.selected} />
          </Tab>
        ),
      },
      ...results.tabs,
    ];

    return results;
  }
}

export default CombatLogParser;
