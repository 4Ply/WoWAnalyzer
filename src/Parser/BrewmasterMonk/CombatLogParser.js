// TODO:

import React from 'react';

import MainCombatLogParser from 'Parser/Core/CombatLogParser';

import SuggestionsTab from 'Main/SuggestionsTab';
import Tab from 'Main/Tab';
import Talents from 'Main/Talents';

// Core
import HealingDone from './Modules/Core/HealingDone';
import DamageTaken from './Modules/Core/DamageTaken';
import DamageDone from './Modules/Core/DamageDone';
import HealingReceived from './Modules/Core/HealingReceived';
import Stagger from './Modules/Core/Stagger';
import IronSkinBrew from './Modules/Spells/IronSkinBrew';

// Features
import CastEfficiency from './Modules/Features/CastEfficiency';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';

class CombatLogParser extends MainCombatLogParser {
  static specModules = {
    // Core
    healingDone: HealingDone,
    healingReceived: HealingReceived,
    damageTaken: DamageTaken,
    stagger: Stagger,
    ironSkinBrew: IronSkinBrew,
    damageDone: DamageDone,
    
    // Features
    alwaysBeCasting: AlwaysBeCasting,
    castEfficiency: CastEfficiency,
    
    // Traits

    // Spells

    // Talents

    // Legendaries / Items:
  };

  generateResults() {
    const results = super.generateResults();

    results.tabs = [
      {
        title: 'Suggestions',
        url: 'suggestions',
        render: () => (
          <SuggestionsTab issues={results.issues} />
        ),
      },
      {
        title: 'Talents',
        url: 'talents',
        render: () => (
          <Tab title="Talents">
            <Talents combatant={this.selectedCombatant} />
          </Tab>
        ),
      },
      ...results.tabs,
    ];
    return results;
  }
}

export default CombatLogParser;
