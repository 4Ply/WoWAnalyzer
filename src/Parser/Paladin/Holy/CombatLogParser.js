import React from 'react';

import Tab from 'Main/Tab';
import Mana from 'Main/Mana';

import CoreCombatLogParser from 'Parser/Core/CombatLogParser';
import LowHealthHealing from 'Parser/Core/Modules/LowHealthHealing';
import HealingDone from 'Parser/Core/Modules/HealingDone';

import LightOfDawnNormalizer from './Normalizers/LightOfDawn';
import DivinePurposeNormalizer from './Normalizers/DivinePurpose';
import BeaconOfVirtueNormalizer from './Normalizers/BeaconOfVirtue';

import PaladinAbilityTracker from './Modules/PaladinCore/PaladinAbilityTracker';
import BeaconHealOriginMatcher from './Modules/PaladinCore/BeaconHealOriginMatcher';
import BeaconTargets from './Modules/PaladinCore/BeaconTargets';
import BeaconHealing from './Modules/PaladinCore/BeaconHealing';
import CastBehavior from './Modules/PaladinCore/CastBehavior';
import Overhealing from './Modules/PaladinCore/Overhealing';
import FillerLightOfTheMartyrs from './Modules/PaladinCore/FillerLightOfTheMartyrs';
import FillerFlashOfLight from './Modules/PaladinCore/FillerFlashOfLight';
import LightOfDawn from './Modules/PaladinCore/LightOfDawn';
import LightOfDawnIndexer from './Modules/PaladinCore/LightOfDawnIndexer';

import Abilities from './Modules/Abilities';
import Checklist from './Modules/Features/Checklist';
import MasteryEffectiveness from './Modules/Features/MasteryEffectiveness';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import CooldownThroughputTracker from './Modules/Features/CooldownThroughputTracker';
import StatValues from './Modules/Features/StatValues';

import RuleOfLaw from './Modules/Talents/RuleOfLaw';
import DevotionAura from './Modules/Talents/DevotionAura';
import AuraOfSacrifice from './Modules/Talents/AuraOfSacrifice';
import AuraOfMercy from './Modules/Talents/AuraOfMercy';
import HolyAvenger from './Modules/Talents/HolyAvenger';
import DivinePurpose from './Modules/Talents/DivinePurpose';
import CrusadersMight from './Modules/Talents/CrusadersMight';
import JudgmentOfLight from './Modules/Talents/JudgmentOfLight';

import { ABILITIES_AFFECTED_BY_HEALING_INCREASES } from './Constants';

class CombatLogParser extends CoreCombatLogParser {
  static abilitiesAffectedByHealingIncreases = ABILITIES_AFFECTED_BY_HEALING_INCREASES;

  static specModules = {
    // Normalizers
    lightOfDawnNormalizer: LightOfDawnNormalizer,
    divinePurposeNormalizer: DivinePurposeNormalizer,
    beaconOfVirtueNormalizer: BeaconOfVirtueNormalizer,

    // Override the ability tracker so we also get stats for IoL and beacon healing
    abilityTracker: PaladinAbilityTracker,
    lowHealthHealing: LowHealthHealing,

    // PaladinCore
    healingDone: [HealingDone, { showStatistic: true }],
    beaconHealOriginMatcher: BeaconHealOriginMatcher,
    beaconTargets: BeaconTargets,
    beaconHealing: BeaconHealing,
    castBehavior: CastBehavior,
    overhealing: Overhealing,
    fillerLightOfTheMartyrs: FillerLightOfTheMartyrs,
    fillerFlashOfLight: FillerFlashOfLight,
    lightOfDawn: LightOfDawn,
    lightOfDawnIndexer: LightOfDawnIndexer,

    // Features
    checklist: Checklist,
    abilities: Abilities,
    masteryEffectiveness: MasteryEffectiveness,
    alwaysBeCasting: AlwaysBeCasting,
    cooldownThroughputTracker: CooldownThroughputTracker,
    statValues: StatValues,

    // Talents
    ruleOfLaw: RuleOfLaw,
    devotionAura: DevotionAura,
    auraOfSacrifice: AuraOfSacrifice,
    auraOfMercy: AuraOfMercy,
    holyAvenger: HolyAvenger,
    divinePurpose: DivinePurpose,
    crusadersMight: CrusadersMight,
    judgmentOfLight: JudgmentOfLight,

    // Items:
  };

  generateResults() {
    const results = super.generateResults();

    // TODO: Suggestion for enchants

    results.tabs = [
      ...results.tabs,
      {
        title: 'Mana',
        url: 'mana',
        render: () => (
          <Tab style={{ padding: '15px 22px' }}>
            <Mana parser={this} />
          </Tab>
        ),
      },
    ];

    return results;
  }
}

export default CombatLogParser;
