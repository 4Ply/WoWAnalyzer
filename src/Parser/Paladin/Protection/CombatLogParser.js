import CoreCombatLogParser from 'Parser/Core/CombatLogParser';
import HealingDone from 'Parser/Core/Modules/HealingDone';
import DamageDone from 'Parser/Core/Modules/DamageDone';
import DamageTaken from 'Parser/Core/Modules/DamageTaken';

import Abilities from './Modules/Abilities';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import SpellUsable from './Modules/Features/SpellUsable';
import Checklist from './Modules/Features/Checklist';

//Spells
import Judgment from './Modules/Spells/Judgment';
import LightOfTheProtectorTiming from './Modules/Features/LightOfTheProtectorTiming';
import ShieldOfTheRighteous from './Modules/Features/ShieldOfTheRighteous';
import Consecration from './Modules/Features/Consecration';

//Items
//import T20_2pc from './Modules/Items/T20_2pc';
//import T20_4pc from './Modules/Items/T20_4pc';
import PillarsOfInmostLight from './Modules/Items/PillarsOfInmostLight';
import TyelcaFerrenMarcussStature from './Modules/Items/TyelcaFerrenMarcussStature';
import BreastplateOfTheGoldenValkyr from './Modules/Items/BreastplateOfTheGoldenValkyr';
import HeadcliffsImmortality from './Modules/Items/HeadcliffsImmortality';

//Talents
import Seraphim from './Modules/Talents/Seraphim';
import RighteousProtector from './Modules/Talents/RighteousProtector';

//import CooldownTracker from './Modules/Features/CooldownTracker';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Core Statistics
    damageTaken: [DamageTaken, { showStatistic: true }],
    damageDone: [DamageDone, { showStatistic: true }],
    healingDone: [HealingDone, { showStatistic: true }],

    // Paladin Core

    // Features
    abilities: Abilities,
    alwaysBeCasting: AlwaysBeCasting,
    spellUsable: SpellUsable,
    checklist: Checklist,
    lightOfTheProtectorTiming: LightOfTheProtectorTiming,
    shieldOfTheRighteous: ShieldOfTheRighteous,
    consecration: Consecration,
    //cooldownTracker: CooldownTracker,

    // Talents
    righteousProtector: RighteousProtector,
    judgment: Judgment,
    seraphim: Seraphim,

    // Items:
    //t20_2pc: T20_2pc,
    //t20_4pc: T20_4pc,
    pillarsOfInmostLight: PillarsOfInmostLight,
    tyelcaFerrenMarcussStature: TyelcaFerrenMarcussStature,
    breastplateOfTheGoldenValkyr: BreastplateOfTheGoldenValkyr,
    headcliffsImmortality: HeadcliffsImmortality,
  };
}

export default CombatLogParser;
