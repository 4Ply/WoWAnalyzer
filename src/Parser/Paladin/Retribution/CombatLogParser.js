import CoreCombatLogParser from 'Parser/Core/CombatLogParser';
import DamageDone from 'Parser/Core/Modules/DamageDone';

import Haste from './Modules/PaladinCore/Haste';

import Abilities from './Modules/Abilities';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import CooldownThroughputTracker from './Modules/Features/CooldownThroughputTracker';
import Checklist from './Modules/Features/Checklist';
import Judgment from './Modules/PaladinCore/Judgment';

import DivinePurpose from './Modules/Talents/DivinePurpose';
import ArtOfWar from './Modules/PaladinCore/ArtOfWar';
import Retribution from './Modules/PaladinCore/Retribution';
import Crusade from './Modules/Talents/Crusade';
import Inquisition from './Modules/Talents/Inquisition';
import RighteousVerdict from './Modules/Talents/RighteousVerdict';

import HolyPowerTracker from './Modules/HolyPower/HolyPowerTracker';
import HolyPowerDetails from './Modules/HolyPower/HolyPowerDetails';

import WhisperOfTheNathrezim from './Modules/Items/WhisperOfTheNathrezim';
import LiadrinsFuryUnleashed from './Modules/Items/LiadrinsFuryUnleashed';
import SoulOfTheHighlord from './Modules/Items/SoulOfTheHighlord';
import AshesToDust from './Modules/Items/AshesToDust';
import ChainOfThrayn from './Modules/Items/ChainOfThrayn';
import Tier20_2set from './Modules/Items/Tier20_2set';
import Tier20_4set from './Modules/Items/Tier20_4set';
import Tier21_2set from './Modules/Items/Tier21_2set';
import Tier21_4set from './Modules/Items/Tier21_4set';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    haste: Haste,
    // PaladinCore
    damageDone: [DamageDone, { showStatistic: true }],
    artOfWar: ArtOfWar,

    // Features
    abilities: Abilities,
    alwaysBeCasting: AlwaysBeCasting,
    cooldownThroughputTracker: CooldownThroughputTracker,
    checklist: Checklist,
    judgment: Judgment,
    retribution: Retribution,
    
    // Talents
    divinePurpose: DivinePurpose,
    crusade: Crusade,
    inquisition: Inquisition,
    righteousVerdict: RighteousVerdict,

    // HolyPower
    holyPowerTracker: HolyPowerTracker,
    holyPowerDetails: HolyPowerDetails,

    // Items:
    whisperOfTheNathrezim: WhisperOfTheNathrezim,
    liadrinsFuryUnleahed: LiadrinsFuryUnleashed,
    soulOfTheHighlord: SoulOfTheHighlord,
    ashesToDust: AshesToDust,
    chainOfThrayn: ChainOfThrayn,
    tier20_2set: Tier20_2set,
    tier20_4set: Tier20_4set,
    tier21_2set: Tier21_2set,
    tier21_4set: Tier21_4set,
  };
}

export default CombatLogParser;
