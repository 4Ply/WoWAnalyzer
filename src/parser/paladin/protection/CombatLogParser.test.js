import integrationTest from 'parser/core/tests/CombatLogParserIntegrationTest';
import CombatLogParser from './CombatLogParser';

const FIGHT_ID = 1;
const PLAYER_ID = 1;

describe('The Protection Paladin Analyzer', integrationTest(
  CombatLogParser,
  'prot-pally-example',
  FIGHT_ID,
  PLAYER_ID
));
