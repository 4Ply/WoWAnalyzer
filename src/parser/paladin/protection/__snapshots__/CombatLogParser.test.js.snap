// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Protection Paladin Analyzer analyzers Abilities matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Protection Paladin Analyzer analyzers AlwaysBeCasting matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt=""
            className="icon"
            src="//render-us.worldofwarcraft.com/icons/56/spell_mage_altertime.jpg"
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Downtime
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Downtime is available time not used to cast anything (including not having your GCD rolling). This can be caused by delays between casting spells, latency, cast interrupting or just simply not casting anything (e.g. due to movement/stunned).<br/>
        <li>You spent <b>65.23%</b> of your time casting something.</li>
        <li>You spent <b>34.77%</b> of your time casting nothing at all.</li>
        "
            >
              34.77 %
            </dfn>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel-footer"
      style={
        Object {
          "borderTop": 0,
          "overflow": "hidden",
          "padding": 0,
        }
      }
    >
      <div
        className="statistic-bar"
      >
        <div
          className="stat-health-bg"
          data-tip="You spent <b>65.23%</b> of your time casting something."
          style={
            Object {
              "width": "65.22732602595921%",
            }
          }
        >
          <img
            alt="Active time"
            src="/img/sword.png"
          />
        </div>
        <div
          className="remainder DeathKnight-bg"
          data-tip="You spent <b>34.77%</b> of your time casting nothing at all."
        >
          <img
            alt="Downtime"
            src="/img/afk.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers ArcaneTorrent matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Protection Paladin Analyzer analyzers Checklist matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Protection Paladin Analyzer analyzers Consecration matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=26573"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/spell_holy_innerfire.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Hits Mitigated w/ Consecration
          </div>
          <div
            className="value"
          >
            70.90 %
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers DamageDone matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt="Sword"
            src="/img/sword.png"
            style={
              Object {
                "border": 0,
              }
            }
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Damage done
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Total damage done: <b>1,696,522</b> "
            >
              5,248 DPS
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers DamageTaken matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt="Shield"
            src="/img/shield.png"
            style={
              Object {
                "border": 0,
              }
            }
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Damage taken
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="The total damage taken was 3,909,576 (0 overkill)."
            >
              12k DTPS
            </dfn>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel-footer"
      style={
        Object {
          "borderTop": 0,
          "overflow": "hidden",
          "padding": 0,
        }
      }
    >
      <div
        className="statistic-bar"
        data-tip="Damage taken by type:
      <ul>
      <li><b>Physical</b>: 1,189,673 (30.43%)</li>
      <li><b>Magic</b>: 2,719,903 (69.57%)</li>
      </ul>
      Damage taken by magic school:
      <ul>
        <li><b>Physical</b>: 1,189,673 (30.43%)</li><li><b>Nature</b>: 2,523,545 (64.55%)</li><li><b>Shadow</b>: 196,358 (5.02%)</li>
      </ul>
      Click the bar to switch between simple and detailed mode."
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <div
          className="spell-school-1-bg"
          style={
            Object {
              "width": "30.429719232980762%",
            }
          }
        />
        <div
          className="spell-school-32-bg"
          style={
            Object {
              "width": "69.57028076701923%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers GrandCrusader matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=85043"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/inv_helmet_74.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Grand Crusader
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Grand Crusader reset the cooldown of Avenger's Shield at least 11 times. 11 are inferred from using it before its cooldown normally be up.<br/>
            You had 83 chances for Grand Crusader to trigger with a 15% chance to trigger."
            >
              11 Resets
            </dfn>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        />
      </div>
      <div
        className="statistic-expansion-button-holster"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-chevron-down"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers HealingDone matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt="Healing"
            src="/img/healing.png"
            style={
              Object {
                "border": 0,
              }
            }
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Healing done
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Total healing done: <b>1,478,097</b>"
            >
              4,572 HPS
            </dfn>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel-footer"
      style={
        Object {
          "borderTop": 0,
          "overflow": "hidden",
          "padding": 0,
        }
      }
    >
      <div
        className="statistic-bar"
      >
        <div
          className="stat-healing-bg"
          data-tip="Regular healing done, not including absorbs. You did a total of 889k regular healing (58.40% of raw healing)."
          style={
            Object {
              "width": "58.404904830229796%",
            }
          }
        >
          <img
            alt="Healing"
            src="/img/healing.png"
          />
        </div>
        <div
          className="stat-healing-absorbs-bg"
          data-tip="Healing by absorbs. This only includes damage prevented by absorbs. You did a total of 461k healing with absorbs (30.33% of raw healing)."
          style={
            Object {
              "width": "30.327144236476126%",
            }
          }
        >
          <img
            alt="Absorb Healing"
            src="/img/absorbed.png"
          />
        </div>
        <div
          className="stat-negative-absorbs-bg"
          data-tip="Absorbed healing. This only includes healing that gets absorbed by debuffs. You had a total of 128k healing absorbed (8.42% of raw healing)."
          style={
            Object {
              "width": "8.419200506888544%",
            }
          }
        >
          <img
            alt="Absorbed Healing"
            src="/img/negative-absorbed.png"
          />
        </div>
        <div
          className="remainder stat-overhealing-bg"
          data-tip="Overhealing. You did a total of 43k overhealing (2.85 % of raw healing). Overhealing can be caused by playing poorly such as selecting the wrong targets or casting abilities at the wrong time, other healers sniping, and/or bringing too many healers."
        >
          <img
            alt="Overhealing"
            src="/img/overhealing.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers InspiringVanguard matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Protection Paladin Analyzer analyzers Judgment matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=53600"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/ability_paladin_shieldofvengeance.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Effective SotR CDR
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Your Judgment casts reduced the cooldown of Shield of the Righteous by <b>106s</b> over 44 casts, 12 of which were critical strikes.
            <br/>
            You wasted <b>0s</b> of cooldown reduction."
            >
              24.69%
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers LightOfTheProtector matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Protection Paladin Analyzer analyzers LightOfTheProtectorTiming matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Protection Paladin Analyzer analyzers MitigationCheck matches the statistic snapshot 1`] = `null`;

exports[`The Protection Paladin Analyzer analyzers RighteousProtector matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Protection Paladin Analyzer analyzers Seraphim matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Protection Paladin Analyzer analyzers ShieldOfTheRighteous matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=53600"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/ability_paladin_shieldofvengeance.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Physical damage mitigated
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Shield of the Righteous usage breakdown:
            <ul>
                <li>You were hit <b>20</b> times with your Shield of the Righteous buff (<b>364,724</b> damage).</li>
                <li>You were hit <b>26</b> times <b><i>without</i></b> your Shield of the Righteous buff (<b>824,949</b> damage).</li>
            </ul>
            <b>43.48%</b> of physical attacks were mitigated with Shield of the Righteous (<b>30.66%</b> of physical damage taken).<br/>
            <b>11</b> of your 32 casts were <em>good</em> (blocked at least 2 melees or a tankbuster, or prevented capping charges)."
            >
              30.66%
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Protection Paladin Analyzer analyzers SpellUsable matches the statistic snapshot 1`] = `"module has no statistic method"`;
